primitives/src/block.rs

```diff
@@ -247,6 +241,37 @@ impl fmt::Debug for Header {
     }
 }
 
+encoding::encoder_newtype! {
+    /// The encoder for the [`Header`] type.
+    pub struct HeaderEncoder(
+        encoding::Encoder6<
+            VersionEncoder,
+            BlockHashEncoder,
+            crate::merkle_tree::TxMerkleNodeEncoder,
+            crate::time::BlockTimeEncoder,
+            crate::pow::CompactTargetEncoder,
+            encoding::ArrayEncoder<4>,
```

Comment:
Just for my understanding, what made you think it better to use full paths here and not import the types?

Reply:
In this instance I'm not fussed because like you say its noisy already and the line count would probably be the same. Just FTR I twitch uncontrollably every time I see `crate::foo::Bar` but that's on me.